src/Endpoints/HttpFactoryTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * HttpFactoryTrait
+ *
+ * @since     Jan 2024
+ * @author    Haydar KULEKCI <haydarkulekci@gmail.com>
+ */
+
+namespace Qdrant\Endpoints;
+
+use Http\Discovery\Psr17FactoryDiscovery;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Qdrant\Exception\InvalidArgumentException;
+
+trait HttpFactoryTrait
+{
+    protected ?RequestFactoryInterface $httpFactory = null;
+
+    public function getHttpFactory(): RequestFactoryInterface
```

Comment:
Why not to require this dependency in AbstractEndpoint constructor?

Reply:
I had not thought that this could be a dependency. Maybe we can move this logic into Qdrant\ClientInterface. It could be the solution. Let's make these changes with the next PR as you suggested. 🙏